### PR TITLE
More doctests and wording fixes for IO stuff

### PR DESCRIPTION
**Pull request by kshyatt**
Just did a super simple example of extending `showerror`. I can add more if people want.

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]
Nice clarification and examples.

File: base/replutil.jl [hunk 1]
```diff
@@ -173,6 +173,24 @@ end
...
+julia> struct MyException <: Exception
+           msg::AbstractString
+       end
+
+julia> function Base.showerror(io::IO, err::MyException)
+           print(io, "MyException: ")
+           print(io, err.msg)
+       end
+
+julia> err = MyException("test exception")
```
Comment (fredrikekre) [inline base/replutil.jl]: Can we do
```julia
julia> throw(MyException("test exception"))
ERROR: MyException: test exception
```
instead? Somewhat more "realistic" case.

Comment (kshyatt) [reply base/replutil.jl]: why not both? i'll push an update (it may not be clear to newbies that `throw` will end up calling `showerror`)

Comment (fredrikekre) [reply base/replutil.jl]: Right, but in normal code you will never call `showerror` yourself, so its a bit misleading.

**Pull request merged**